### PR TITLE
common/perf_counters: make schema more friendly and update docs

### PR DESCRIPTION
**Pull request by liewegas**
I just spoke with a user yesterday who was complaining about the lack of documentation for the metrics.  (The docs exist, but they're stale; when I showed him the ideally self-documenting  'perf schema' output it wasn't obvious what was a guage or metric.)

### CODE REVIEW

**Comment by liewegas** [comment]
@branch-predictor Do you mind submitting a patch to the doc file similarly documenting the 'perf historgram (schema|dump)' commands?

**Comment by branch-predictor** [comment]
@liewegas sure, on it.

**Comment by liewegas** [comment]
thank you!

**Comment by branch-predictor** [comment]
@liewegas https://github.com/ovh/ceph/tree/bp-doc-perf-histograms feel free to pull that in.

**Comment by liewegas** [comment]
updated

**Comment by badone** [comment]
@liewegas there is a failing test (unittest_perf_counters), see my comment above.

File: src/common/perf_counters.cc [hunk 1]
```diff
@@ -497,14 +521,14 @@ void PerfCountersBuilder::add_time_avg(
...
-void PerfCountersBuilder::add_histogram(
+void PerfCountersBuilder::add_u64_counter_histogram(
   int idx, const char *name,
   PerfHistogramCommon::axis_config_d x_axis_config,
   PerfHistogramCommon::axis_config_d y_axis_config,
   const char *description, const char *nick, int prio)
 {
-  add_impl(idx, name, description, nick, prio,
-	   PERFCOUNTER_U64 | PERFCOUNTER_HISTOGRAM,
+  add_impl(idx, name, description, nick, prior,
```
Comment (badone) [inline src/common/perf_counters.cc]: Should be prio?

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-05-16_22:22:40-rados-wip-sage-testing2---basic-smithi/

**Pull request merged**